families/supplychain/restapi/openapi.yaml

```diff
@@ -0,0 +1,398 @@
+
+paths:
+  /agents:
+    get:
+      summary: Fetches the data for the current set of Agents
+      description: >
+        Fetches a paginated list of all Agents. Using the `identifier` or name
+        filter parameter, will narrow the list to any Agents that have an
+        identifier or name respectively containing the specified string.
+
+        Agents are identified by their public keys.
+      parameters:
+        - name: identifier
+          in: query
+          type: string
+          description: A partial public_key to filter Agents by
+        - name: name
+          in: query
+          type: string
+          description: A partial name to filter Agents by
+        - $ref: "#/parameters/count"
+        - $ref: "#/parameters/min"
+        - $ref: "#/parameters/max"
+        - $ref: "#/parameters/sort"
+      responses:
+        200:
+          description: Successfully retrieved Agents
+          schema:
+            properties:
+              data:
+                type: array
+                items:
+                  $ref: "#/definitions/Agent"
+              link:
+                $ref: "#/definitions/Link"
+              paging:
+                $ref: "#/definitions/Paging"
+        400:
+          $ref: "#/responses/400BadRequest"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+  /agents/{agent_id}:
+    parameters:
+      - $ref: "#/parameters/agent_id"
+    get:
+      summary: Fetches a particular Agent.
+      responses:
+        200:
+          description: Successfully fetched Agent
+          schema:
+            properties:
+              data:
+                $ref: "#/definitions/Agent"
+              link:
+                $ref: "#/definitions/Link"
+        400:
+          $ref: "#/responses/400BadRequest"
+        404:
+          $ref: "#/responses/404NotFound"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+  /applications:
+    get:
+      summary: Fetches the data for the current set of Applications
+      description: >
+        Fetches a paginated list of all Applications. Using the `applicant`
+        parameter, will narrow the list to any Applications that have a
+        matching value in their applicant field. Using the `identifier` filter
+        parameter, will narrow the list to any Applications that have an
+        identifier containing the specified string. Using the `state`
+        parameter, will narrow the list to any Applications that have a
+        matching value in their state field.
+
+      parameters:
+        - name: applicant
+          in: query
+          type: string
+          description: The public key of the applicatant Agent to filter by
+        - name: identifier
+          in: query
+          type: string
+          description: A partial identifier to filter Applications by.
```

Comment:
Do partial identifiers still work as filters in this case (and the others), or is this partial idea a hold-over from cloning this file from the base Sawtooth REST API spec?